adobe_vipm/flows/migration.py

```diff
@@ -159,110 +159,138 @@ def populate_offers_for_transfer(product_id, transfer, transfer_preview):
     create_offers(product_id, offers)
 
 
+def handle_preview_error(transfer, api_err):
+    """Handle Adobe API errors during transfer preview"""
+    if api_err.code == STATUS_TRANSFER_ALREADY_TRANSFERRED:
+        return True
+
+    transfer.adobe_error_code = api_err.code
+    transfer.adobe_error_description = str(api_err)
+
+    if any(x in str(api_err) for x in RECOVERABLE_TRANSFER_ERRORS):
+        transfer.status = "rescheduled"
+        transfer.migration_error_description = (
+            "Adobe transient error received during transfer preview.")
+        check_reschedules(transfer)
+    else:
+        transfer.status = "failed"
+        transfer.migration_error_description = "Adobe error received during transfer preview."
+        transfer.updated_at = datetime.now()
+        transfer.save()
+        send_exception(
+            "Adobe error received during transfer preview.",
+            "An unexpected error has been received from Adobe asking for preview "
+            f"of transfer for Membership **{transfer.membership_id}**.",
+            facts=FactsSection(
+                "Last error from Adobe",
+                {transfer.adobe_error_code: transfer.adobe_error_description},
+            ),
+            button=get_transfer_link_button(transfer),
+        )
+    return False
+
+def handle_preview_authorization_error(transfer, e):
+    """Handle authorization errors during transfer preview"""
+    transfer.status = "failed"
+    transfer.migration_error_description = str(e)
+    transfer.updated_at = datetime.now()
+    transfer.save()
+    send_exception(
+        "Marketplace Platform configuration error during transfer.",
+        str(e),
+        facts=FactsSection(
+            "Transfer error",
+            {"AuthorizationNotFoundError": transfer.migration_error_description},
+        ),
+        button=get_transfer_link_button(transfer),
+    )
+
+def handle_transfer_creation_error(transfer, api_err):
+    """Handle Adobe API errors during transfer creation"""
+    transfer.adobe_error_code = api_err.code
+    transfer.adobe_error_description = str(api_err)
+    transfer.migration_error_description = "Adobe error received during transfer creation."
+    transfer.status = "failed"
+    transfer.updated_at = datetime.now()
+    transfer.save()
+    send_exception(
+        "Adobe error received during transfer creation.",
+        "An unexpected error has been received from Adobe creating the "
+        f"transfer for Membership **{transfer.membership_id}**.",
+        facts=FactsSection(
+            "Last error from Adobe",
+            {transfer.adobe_error_code: transfer.adobe_error_description},
+        ),
+        button=get_transfer_link_button(transfer),
+    )
+
+def handle_transfer_reseller_error(transfer, e):
+    """Handle reseller errors during transfer creation"""
```

Comment:
I see some repetitive pattern here when you fail transfer, like when you fail with Adobe Error or not